src/ml_thread.py

```diff
@@ -68,12 +68,12 @@ def minibatch_backprop_and_update_weights(self, minibatch_idx, images, labels):
         self.optimizer.zero_grad()
         loss.backward()
 
-        # Update weights on this minibatch's gradient
-        self.optimizer.step()
-
         # Get gradients for this minibatch
         minibatch_updates = { idx: params.clone() for idx, params in self.parameter_pointers.items() }
```

Comment:
I thought you said you would do `params.grad.clone()` instead of `params.clone()`?

Reply:
You're right. Here's the issue: When I run it with `params.grad.clone()` we run into a division by zero error in the linear programming bit on one of the receivers who tries to aggregate. Here's the stack trace. I don't know how to solve it
```
AGG FROM:  localhost:5000
AGG FROM:  localhost:5001
Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.7.2_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python/3.7.2_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "main.py", line 22, in _actually_run
    self.node.train()
  File "/Users/weitaiting/Documents/snlab/gaia2/src/ml_thread.py", line 158, in train
    self.aggregate_received_updates()
  File "/Users/weitaiting/Documents/snlab/gaia2/src/ml_thread.py", line 130, in aggregate_received_updates
    alphas = self.fairness_state.get_alphas(flattened_metadata_list)
  File "/Users/weitaiting/Documents/snlab/gaia2/src/update_metadata/device_fairness.py", line 57, in get_alphas
    return get_weights(v)
  File "/Users/weitaiting/Documents/snlab/gaia2/src/get_weights.py", line 15, in get_weights
    x = 1.0 / l
ZeroDivisionError: float division by zero
```